src/photon/photon_scheduler.c

```diff
@@ -110,22 +110,16 @@ void free_local_scheduler(local_scheduler_state *state) {
 
 void assign_task_to_worker(local_scheduler_state *state,
                            task_spec *spec,
-                           int worker_index,
-                           bool from_global_scheduler) {
+                           int worker_index) {
   CHECK(worker_index < utarray_len(state->workers));
   worker *w = (worker *) utarray_eltptr(state->workers, worker_index);
   write_message(w->sock, EXECUTE_TASK, task_spec_size(spec), (uint8_t *) spec);
   /* Update the global task table. */
   if (state->db != NULL) {
     task *task =
         alloc_task(spec, TASK_STATUS_RUNNING, get_db_client_id(state->db));
-    if (from_global_scheduler) {
-      task_table_update(state->db, task, (retry_info *) &photon_retry, NULL,
-                        NULL);
-    } else {
-      task_table_add_task(state->db, task, (retry_info *) &photon_retry, NULL,
-                          NULL);
-    }
+    task_table_update(state->db, task, (retry_info *) &photon_retry, NULL,
```

Comment:
We're now calling `task_table_add_task` twice for every task that is assigned locally, whereas before it was just once.